### PR TITLE
ON-16333 # Added emailTemplate types

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: typescript/submissionEvents.d.ts [hunk 1]
```diff
@@ -28,6 +27,17 @@ export type CallbackSubmissionEvent = FormSubmissionEventConditional & {
   isDraft: boolean
 }
 
+type PdfSubmissionEventEmailTemplate =
+  | {
+      mustacheTag: string
+      type: 'FORM_ELEMENT'
```
Comment (mymattcarroll) [inline typescript/submissionEvents.d.ts:33]: There are a bunch of others we need to support as well. See the AC.

File: typescript/submissionEvents.d.ts [hunk 2]
```diff
@@ -37,6 +47,10 @@ export type PdfSubmissionEvent = FormSubmissionEventConditional & {
     includeSubmissionIdInPdf?: boolean
     excludedElementIds?: string[]
     usePagesAsBreaks?: boolean
+    emailTemplate?: {
```
Comment (kizaonline) [inline typescript/submissionEvents.d.ts:50]: Should we also add it to other submission event types that produce PDF's? e.g. TRIM, Civica...

Comment (mymattcarroll) [reply typescript/submissionEvents.d.ts:50]: This is not PDF config, its email config.

File: typescript/submissionEvents.d.ts (hunk 1 above)
Comment (Zaxist) [reply typescript/submissionEvents.d.ts:33]: Wondering whether those should just be replaceable values inside the `text`
```
"Blah blah blah {EXTERNAL_ID} blah blah"
```

**Pull request merged**